src/inner.jl

```diff
@@ -0,0 +1,143 @@
+using ITensors: inner, scalar, loginner
+using LinearAlgebra: norm, norm_sqr
+
+default_algorithm(tns::Tuple) = "bp"
```

Comment:
I think this should be `default_contract_alg` to be more specific and also match the style that we use `alg` for the algorithm keyword argument.